advanced-cache.php

```diff
@@ -334,6 +335,21 @@ function add_debug_html_to_output( $debug_html ) {
 		}
 		$this->cache['output'] .= "\n$debug_html";
 	}
+
+	function set_query( $query_string ) {
+		parse_str( $query_string, $this->query );
+
+		foreach ( $this->ignored_query_args as $arg ) {
+			unset( $this->query[ $arg ] );
+		}
+
+		if ( empty( $this->query ) ) {
+			$this->query = ''; // Because this is the default $query value.
```

Comment:
This line changes the previous behavior. In my testing, `parse_str()` sets `$query` to an array even if there is nothing to parse, and the previous code did nothing to set `$query` back to a string.

I'm actually wondering whether it would be better to initially set `$query` to an empty array for consistency, but it's probably best to leave it alone.

Reply:
Only 1 reason. The initial `$query = ''`. If the `$_SERVER['QUERY_STRING']` is set, but we remove all keys, you end up having 2 different cache keys. For `$query = ''` and for `$query = []`.

That said, I can probably switch the default for `$query = []`.

Reply:
Actually, I'd rather leave it. Thinking of doing that on WP.com scares me, as it will invalidate all the homepage caches. :)

Reply:
I think it is probably fine to reset in that case, but it would be good to add a comment explaining the condition you were trying to avoid (i.e., 2 different cache keys).

I don't have a strong opinion on the default.

Also, what if you add an early return for an empty query string?

Reply:
> Also, what if you add an early return for an empty query string?

Hmm, don't think we can, have to check after the ignored args are unset.

Reply:
Updated the `$query = array()`. Should be consistent, and no duplicate keys.